src/features/wallet/send/SendTransferForm.tsx

```diff
@@ -0,0 +1,420 @@
+
+  // Set address alias if necessary
+  useAsync(async () => {
+    if (address.match(/.*\.eth$/)) {
+      setAddressLoading(true)
+      const { address: ensAddress } = await ensLookup(address)
+      if (ensAddress) {
+        setAddressAlias(address)
+        setAddress(ensAddress)
+        setAddressLoading(false)
+        return
+      }
+    }
+    setAddressLoading(false)
+    setAddressAlias('')
+  }, [address])
+
+  // Update the internal HNT amount based on form input
+  useEffect(() => {
+    const hntBalance = Balance.fromFloat(
+      parseFloat(amount),
+      CurrencyType.networkToken,
+    )
+    setBalanceAmount(hntBalance)
+  }, [amount])
+
+  // Update the required fee if HNT amount changes
+  useAsync(async () => {
+    updateFee()
+  }, [balanceAmount, transferData?.amountToSeller])
+
+  const getNonce = (): number => {
+    if (!account?.speculativeNonce) return 1
+    return account.speculativeNonce + 1
+  }
+
+  const updateFee = async () => {
+    let dcFee
+    if (type === 'payment') {
+      dcFee = await calculatePaymentTxnFee(
+        balanceAmount.integerBalance,
+        getNonce(),
+        address,
+      )
+    } else if (type === 'dc_burn') {
+      dcFee = await calculateBurnTxnFee(
+        balanceAmount.integerBalance,
+        address,
+        getNonce(),
+        memo,
+      )
+    } else if (type === 'transfer') {
+      dcFee = await calculateTransferTxnFee(transferData?.partialTransaction)
+    } else {
+      throw new Error('Unsupported transaction type')
+    }
+    const hntFee = feeToHNT(dcFee)
+    setFee(hntFee)
+  }
+
+  // Helper to normalize direct "amount" input value
+  const setFormAmount = (formAmount: string) => {
+    if (formAmount === decimalSeparator || formAmount.includes('NaN')) {
+      return setAmount(`0${decimalSeparator}`)
+    }
+    const rawInteger = (formAmount.split(decimalSeparator)[0] || formAmount)
+      .split(groupSeparator)
+      .join('')
+    const integer = parseInt(rawInteger, 10).toLocaleString(locale)
+    let decimal = formAmount.split(decimalSeparator)[1]
+    if (integer === 'NaN') {
+      return setAmount('')
+    }
+    if (decimal && decimal.length >= 9) decimal = decimal.slice(0, 8)
+    setAmount(
+      formAmount.includes(decimalSeparator)
+        ? `${integer}${decimalSeparator}${decimal}`
+        : integer,
+    )
+  }
+
+  const setMaxAmount = () => {
+    triggerNavHaptic()
+    const balance = account?.balance
+    if (!balance) return
+    if (fee > balance) {
+      const balanceStr = balance.toString(8, {
+        decimalSeparator,
+        groupSeparator,
+        showTicker: false,
+      })
+      setAmount(balanceStr)
+    } else {
+      const maxAmount = balance.minus(fee)
+      const maxAmountStr = maxAmount.toString(8, {
+        decimalSeparator,
+        groupSeparator,
+        showTicker: false,
+      })
+      setAmount(maxAmountStr)
+    }
+  }
+
+  const renderLockedPaymentForm = () => [
```

Comment:
This shouldn't return an array, it should use `(` and `)` instead of `[` and `]`

Reply:
For this comment (and the related ones within this file), is this stylistic feedback? Or is there a functional purpose? Given all form elements are nested within a `<Box />` in the component's return value, I figured we didn't need to wrap them again within another `<Box />` (or React fragment) in these internal helpers.

Reply:
I actually really like this syntax, my concern is that I'm pretty sure it will create [this](https://reactjs.org/docs/lists-and-keys.html#keys) warning because arrays need to have keys.

Typically we've been just wrapping in fragments like this 
```
const renderLockedPaymentForm = () => (
  <>
    <Component />
    <OtherComponent />
  </>
)

but, you could just give each component a key and keep this current style.

Reply:
I think for fixed outputs a fragment is the expected element, arrays should be used for more dynamic content. More info on fragments can be found here: https://reactjs.org/docs/fragments.html

Reply:
Got it, sounds like fragments are the way to go here, updated: https://github.com/helium/hotspot-app/pull/479/commits/9e2cfb92dc60f3f8dfa082c4641520c6c2fd4cb7?w=1